### PR TITLE
Optimize cloudcare api calls

### PR DESCRIPTION
**Pull request by gcapalbo**
@czue @benrudolph 

Since the touchforms api user is making the calls to get case info when starting up smsforms sessions, any time we send a survey to a case (i.e., no user_id filter) with footprint=true, it queries and returns info for all the cases in the domain.  This PR makes the queries more targeted so that we don't use footprint=true for about 99% of the situations where we start smsforms sessions and there's no user_id. 

We still need to include footprint=true for a very small number of use cases (when a form is being filled out by a subcase and it references the parent case).  @czue see comment in this PR about a possible way to optimize that further.

cc @nickpell code review buddy

### CODE REVIEW

File: corehq/apps/cloudcare/views.py [hunk 1]
```diff
@@ -264,8 +263,7 @@ def get_cases(request, domain):
     ids_only = string_to_boolean(request.REQUEST.get("ids_only", "false"))
     case_id = request.REQUEST.get("case_id", "")
     footprint = string_to_boolean(request.REQUEST.get("footprint", "false"))
-    include_children = string_to_boolean(request.REQUEST.get("include_children", "false"))
-    if case_id and not footprint and not include_children:
+    if case_id and not footprint:
```
Comment (gcapalbo) [inline corehq/apps/cloudcare/views.py]: @czue Do you know why we check for `not footprint` here? If we could also allow returning all the case's ancestors when we specify a case_id filter, we can optimize the remaining 1% of use cases I was referring to.

File: corehq/apps/cloudcare/views.py [hunk 2]
```diff
@@ -228,7 +228,6 @@ def get_cases_vary_on(request, domain):
         request.REQUEST.get('ids_only', 'false'),
         request.REQUEST.get('case_id', ''),
         request.REQUEST.get('footprint', 'false'),
-        request.REQUEST.get('include_children', 'false'),
```
Comment (gcapalbo) [inline corehq/apps/cloudcare/views.py]: @benrudolph Is removing this going to confuse anything for the cache you setup?

Comment (benrudolph) [inline corehq/apps/cloudcare/views.py]: nope should be fine

**Comment by benrudolph** [comment]
looks good to me. will wait for this to be answered cause i'm not quite sure: https://github.com/dimagi/commcare-hq/pull/7638/files#r35536662. one more step towards a fast cloudcare! :walking:

File: corehq/apps/cloudcare/views.py (hunk 1 above)
Comment (czue) [inline corehq/apps/cloudcare/views.py]: @gcapalbo git blame led me here: https://github.com/dimagi/commcare-hq/pull/621/ - is that fix/change no longer necessary?

**Comment by czue** [comment]
looks good to me.

**Pull request merged**